lib/modal/edit_product_modal.dart

```diff
@@ -0,0 +1,358 @@
+                        labelText: "Unit",
+                      ),
+                    ),
+                    popupProps: const PopupProps.menu(
+                      fit: FlexFit.loose,
+                    ),
+                    validator: (value) {
+                      if (value == null || value.isEmpty) {
+                        return '';
+                      }
+
+                      if (errors.containsPartialKey('unit')) {
+                        return errors[errors.getValue('unit')][0];
+                      }
+                      return null;
+                    },
+                  ),
+                  const SizedBox(
+                    height: 10,
+                  ),
+                  TextFormField(
+                    decoration: const InputDecoration(
+                      hintText: 'Stock',
+                      labelText: 'Stock',
+                    ),
+                    controller: stockController,
+                    keyboardType: TextInputType.number,
+                    inputFormatters: [
+                      FilteringTextInputFormatter.digitsOnly,
+                    ],
+                    validator: (value) {
+                      if (value == null || value.isEmpty) {
+                        return '';
+                      }
+                      // if value is above 65536
+                      if (int.tryParse(value) == null ||
+                          int.tryParse(value)! > 65535) {
+                        return 'Stock must be under 65535';
+                      }
+
+                      if (errors.containsPartialKey('stock')) {
+                        return errors[errors.getValue('stock')][0];
+                      }
+                      return null;
+                    },
+                  ),
+                  const SizedBox(
+                    height: 10,
+                  ),
+                  TextFormField(
+                    decoration: const InputDecoration(
+                      hintText: 'Shelf',
+                      labelText: 'Shelf',
+                    ),
+                    controller: shelfController,
+                    keyboardType: TextInputType.number,
+                    inputFormatters: [
+                      FilteringTextInputFormatter.digitsOnly,
+                    ],
+                    validator: (value) {
+                      if (value == null || value.isEmpty) {
+                        return '';
+                      }
+
+                      if (int.tryParse(value) == null ||
+                          int.tryParse(value)! > 65535) {
+                        return 'Shelf must be under 65535';
+                      }
+
+                      if (errors.containsPartialKey('shelf')) {
+                        return errors[errors.getValue('shelf')][0];
+                      }
+                      return null;
+                    },
+                  ),
+                ],
+              ),
```

Comment:
Low priority. Could change stock and shelf amount based on location chosen

Reply:
No, I will make a seperate story for this :)